Lib/email/utils.py

```diff
@@ -106,12 +106,42 @@ def formataddr(pair, charset='utf-8'):
     return address
 
 
+def _pre_parse_validation(fieldvalues):
+    """Validate the field values are syntactically correct"""
+    for v in fieldvalues:
+        s = str(v).replace('\\(', '').replace('\\)', '')
+        if s.count('(') != s.count(')'):
+            fieldvalues.remove(v)
+            fieldvalues.append('')
+
+    return fieldvalues
+
+
+def _post_parse_validation(parsedvalues):
+    """Validate the parsed values are syntactically correct"""
+    for v in parsedvalues:
+        if '[' in v[1]:
+            parsedvalues.remove(v)
+            parsedvalues.append(('', ''))
+
+    return parsedvalues
+
 
 def getaddresses(fieldvalues):
     """Return a list of (REALNAME, EMAIL) for each fieldvalue."""
+    fieldvalues = _pre_parse_validation(fieldvalues)
     all = COMMASPACE.join(str(v) for v in fieldvalues)
```

Comment:
The existing code already calls `str(v)` here... this is a problem.  It means the input can be anything.  But we probably cannot change it in a patch release for a security fix.

To avoid propagating this mistake into more lines of code spread out, I suggest changing this function to do:

```python
fieldvalues = [str(v) for v in fieldvalues]
```

on the first line and get rid of all subsequent str(v) calls on anything from fieldvalues in this function or in functions it calls.